applications/PfemFluidDynamicsApplication/custom_processes/adaptive_time_interval_process.hpp

```diff
@@ -284,148 +284,160 @@ namespace Kratos
-		    updatedEnlargedElementCoordinates.push_back(Node<3>::Pointer(new Node<3>(i,updatedNodalCoordinates.X(),updatedNodalCoordinates.Y(),updatedNodalCoordinates.Z())));
+		    Point updatedNodalCoordinates=itElem->GetGeometry()[i].Coordinates()+Vel*temporaryTimeInterval;
+		    updatedElementCoordinates.push_back(Node<3>::Pointer(new Node<3>(i,updatedNodalCoordinates.X(),updatedNodalCoordinates.Y(),updatedNodalCoordinates.Z())));
 		  }
 
+		double newVolume=0;
 		if(itElem->GetGeometry().size()==4){
-		  Tetrahedra3D4<Node<3> > myGeometry(updatedEnlargedElementCoordinates);
+		  Tetrahedra3D4<Node<3> > myGeometry(updatedElementCoordinates);
 		  newVolume=myGeometry.Volume();
 		}else if(itElem->GetGeometry().size()==10){
-		  Tetrahedra3D10<Node<3> > myGeometry(updatedEnlargedElementCoordinates);
+		  Tetrahedra3D10<Node<3> > myGeometry(updatedElementCoordinates);
 		  newVolume=myGeometry.Volume();
 		}else{
 		  std::cout<<"GEOMETRY NOT DEFINED"<<std::endl;
 		}
 
-		if(newVolume<0.001*currentElementalVolume && currentElementalVolume>0){
-		  increaseTimeInterval=false;
-		  /* std::cout<<"I'll not reduce the time step but I'll not allow to increase it"<<std::endl; */
+		if(solidElement==true){
+		  newVolume=currentElementalVolume;
 		}
 
+		if(newVolume<0.001*currentElementalVolume && currentElementalVolume>0){
+		  double reducedTimeInterval=0.5*temporaryTimeInterval;
+	      
+		  if(reducedTimeInterval<temporaryTimeInterval){
+		    rCurrentProcessInfo.SetValue(DELTA_TIME,reducedTimeInterval);
+		    /* std::cout<<"reducing time step (elemental inversion)"<<reducedTimeInterval<<std::endl; */
+		    rCurrentProcessInfo.SetValue(TIME_INTERVAL_CHANGED,true);
+		    increaseTimeInterval=false;
+		    break;
+		  }
+		}else{
+		  Geometry<Node<3> >  updatedEnlargedElementCoordinates;
+
+		  for(unsigned int i=0; i<itElem->GetGeometry().size(); i++)
+		    {
+		      const array_1d<double,3> &Vel = itElem->GetGeometry()[i].FastGetSolutionStepValue(VELOCITY);
+		      Point updatedNodalCoordinates=itElem->GetGeometry()[i].Coordinates()+Vel*temporaryTimeInterval*2.5;
+		      updatedEnlargedElementCoordinates.push_back(Node<3>::Pointer(new Node<3>(i,updatedNodalCoordinates.X(),updatedNodalCoordinates.Y(),updatedNodalCoordinates.Z())));
+		    }
+
+		  if(itElem->GetGeometry().size()==4){
+		    Tetrahedra3D4<Node<3> > myGeometry(updatedEnlargedElementCoordinates);
+		    newVolume=myGeometry.Volume();
+		  }else if(itElem->GetGeometry().size()==10){
+		    Tetrahedra3D10<Node<3> > myGeometry(updatedEnlargedElementCoordinates);
+		    newVolume=myGeometry.Volume();
+		  }else{
+		    std::cout<<"GEOMETRY NOT DEFINED"<<std::endl;
+		  }
+
+		  if(newVolume<0.001*currentElementalVolume && currentElementalVolume>0){
+		    increaseTimeInterval=false;
+		    /* std::cout<<"I'll not reduce the time step but I'll not allow to increase it"<<std::endl; */
+		  }
 
-	      }
+
+		}
 
 
 
+	      }
 	    }
```

Comment:
Why do you want to modify this file? Are these the same changes that crashed the code some days ago?

Reply:
I hope the merge is done correctly. The changes were done to allow to print walls in GiD output. It was interferring in this process and this correction was needed.

Reply:
This function is for reducing adaptively the time step, I do not see how it can interfere with the GiD output. Anyway, there are changes of the function aim that I do not understand. Please, let the previous version of the function and, if required, we will modify in a new specific pull request.

Reply:
Please try to comment the new lines and test the behaviour. Then you will understand why they are needed.

Reply:
we will discuss it the next week in the office, now I cannot test everything...

Reply:
To keep the same behaviour of the process wall elements must not be considered in the loop. This new elements are there for writing purposes. The original implementation is not aware of this matter and without this change the process fails.